appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -150,6 +154,26 @@
      */
     public ApplicationProcessor(Set<Class<?>> appClasses) {
         this.classes = appClasses;
+        this.classes.removeIf(cls -> cls.isInterface() || Modifier.isAbstract(cls.getModifiers()));
+        if (classes != null) {
+            addInnerClasses();
```

Comment:
FYI: Added this here as an after-fact since it allows to only have one place that both works in full integration setting or production setting as well as our test setup. This makes this effective in our testing and keeps the test setup in line what will happen in production.